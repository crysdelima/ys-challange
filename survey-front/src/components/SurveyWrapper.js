import React from 'react';
import PropTypes from 'prop-types';
import { Button, Modal, Step, Segment } from 'semantic-ui-react';

const SurveyWrapper = ({
    children,
    disableNextStep,
    hasPrevStep,
    hasNextStep,
    listWithActiveStep,
    onClickNextStep,
    onClickPrevStep,
    onSubmit,
    openModal,
}) => {
    return (
        <Modal
            centered={false}
            open={openModal}
        >
            <Modal.Header>
                Profile Survey
            </Modal.Header>

            <Modal.Content>
                {listWithActiveStep.length > 0 && (
                    <Step.Group
                        attached="top"
                        items={listWithActiveStep}
                        widths={listWithActiveStep.length}
                    />
                )}

                <Segment attached>
                    {children}
                </Segment>
            </Modal.Content>

            <Modal.Actions>
                {hasPrevStep && (
                    <Button
                        content="Previous"
                        icon='left arrow'
                        labelPosition='left'
                        onClick={onClickPrevStep}
                        primary
                    />
                )}

                {hasNextStep ? (
                    <Button
                        content="Next"
                        disabled={disableNextStep}
                        icon='right arrow'
                        labelPosition='right'
                        onClick={onClickNextStep}
                        primary
                    />
                ) : (
                    <Button
                        content="Submit"
                        icon='right arrow'
                        labelPosition='right'
                        onClick={onSubmit}
                        color="green"
                    />
                )}
            </Modal.Actions>
        </Modal>
    )
};

SurveyWrapper.propTypes = {
    children: PropTypes.element,
    disableNextStep: PropTypes.bool.isRequired,
    hasNextStep: PropTypes.bool,
    hasPrevStep: PropTypes.bool,
    listWithActiveStep: PropTypes.array,
    onClickNextStep: PropTypes.func.isRequired,
    onClickPrevStep: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    openModal: PropTypes.bool.isRequired,
};

SurveyWrapper.defaultProps = {
    children: null,
    hasNextStep: null,
    hasPrevStep: null,
    listWithActiveStep: [],
}

export default SurveyWrapper;
