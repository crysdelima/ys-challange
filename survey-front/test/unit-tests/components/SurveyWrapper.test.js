import { shallow } from 'enzyme';
import SurveyWrapper from '../../../src/components/SurveyWrapper';
import toJson from 'enzyme-to-json';

const steps = [{
    key: 'step1',
    icon: 'table',
    title: 'Step 1'
}, {
    key: 'step2',
    icon: 'info',
    title: 'Step 2',
}];

describe('<SurveyWrapper />', () => {
    test('Render correctly', () => {
        const wrapper = shallow(
            <SurveyWrapper
                disableNextStep={false}
                listWithActiveStep={steps}
                onClickPrevStep={() => {}}
                onClickNextStep={() => {}}
                onSubmit={() => {}}
                openModal={true}
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('Should correctly handle navigation buttons', () => {
        const onClickPrevStep = jest.fn();
        const onClickNextStep = jest.fn();
        const onSubmit = jest.fn();
        
        const wrapper = shallow(
            <SurveyWrapper
                disableNextStep={false}
                listWithActiveStep={steps}
                onClickPrevStep={onClickPrevStep}
                onClickNextStep={onClickNextStep}
                onSubmit={onSubmit}
                openModal={true}
            />
        );
        // expect(toJson(wrapper)).toMatchSnapshot();
    });
})
