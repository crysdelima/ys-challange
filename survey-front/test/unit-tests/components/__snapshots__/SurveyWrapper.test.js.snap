// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SurveyWrapper /> Render correctly 1`] = `
<Modal
  centered={false}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  open={true}
>
  <ModalHeader>
    Profile Survey
  </ModalHeader>
  <ModalContent>
    <StepGroup
      attached="top"
      items={
        Array [
          Object {
            "icon": "table",
            "key": "step1",
            "title": "Step 1",
          },
          Object {
            "icon": "info",
            "key": "step2",
            "title": "Step 2",
          },
        ]
      }
      widths={2}
    />
    <Segment
      attached={true}
    />
  </ModalContent>
  <ModalActions>
    <Button
      as="button"
      content="Previous"
      icon="left arrow"
      labelPosition="left"
      onClick={[Function]}
      primary={true}
    />
    <Button
      as="button"
      content="Next"
      disabled={false}
      icon="right arrow"
      labelPosition="right"
      onClick={[Function]}
      primary={true}
    />
    <Button
      as="button"
      color="green"
      content="Submit"
      icon="right arrow"
      labelPosition="right"
      onClick={[Function]}
    />
  </ModalActions>
</Modal>
`;
